doc: update for NetBSD 9.2, add GUI Build Instructions

jarolrod opened the pull request:
**For reviewer:** as I suppose few have a NetBSD system available, I wrote a [guide](https://gist.github.com/jarolrod/385dc063bb02c90aea0cbe8a147fc418#file-netbsd-vm-setup-guide-md) to setup a VM for testing purposes.

This attempts to update the NetBSD docs so one can successfully build on the latest release. It also adds instructions to build the GUI.

Additionally, it includes a note and an example on how one could update the gcc version bundled with NetBSD 9.2 and prior to be able to actually compile. This note can be updated with the release of NetBSD 10, as it will package an acceptable gcc version.

Master: [render](https://github.com/bitcoin/bitcoin/blob/master/doc/build-netbsd.md)
PR: [render](https://github.com/bitcoin/bitcoin/blob/d3e9a1c71bef1d730b5820f85e9758af54267ac3/doc/build-netbsd.md)

Related to #20610, but reworked.

jarolrod committed:
doc: update for NetBSD 9.2, add GUI Build Instructions

NetBSD doc has not seen any meaningful contribution since 2018.
This PR intends to update the docs so that one can successfully build on
the latest NetBSD release. It also adds dependency information and
instructions to build the GUI.

fanquake commented on doc/build-netbsd.md:
```diff
@@ -1,92 +1,144 @@
+It is not necessary to build wallet functionality to run bitcoind or the GUI.
 
-#### With descriptor wallet:
+###### Legacy Wallet Support
```
Please leave the descriptor wallet section first.

fanquake commented on doc/build-netbsd.md:
```diff
@@ -1,92 +1,144 @@
-#### With descriptor wallet:
+###### Legacy Wallet Support
+
+`db4` is required to enable support for legacy wallets. Skip if you don't intend to use legacy wallets.
```
> Skip if you don't intend to use legacy wallets.

I think sentences like this (repeated for every dependency) are overly verbose (and somewhat obvious) and could be dropped.

fanquake commented on doc/build-netbsd.md:
```diff
@@ -1,92 +1,144 @@
+There are many ways to configure Bitcoin Core. Here are a few common examples:
+##### Wallet (BDB + SQlite) Support, No GUI:
+
+This explicitly enables legacy wallet support and disables the GUI. An error will be thrown if `db4` is not installed. If `sqlite3` is installed, then descriptor wallet support will also be built.
```
> An error will be thrown if `db4` is not installed.

I really don't this we need to say things like this in this document.

I would rather drop all these example configuration examples (maybe except for one), and tell users they can look at the output of ./configure --help.

jarolrod replied:
true, i left the example where the wallet and gui are explicitly disabled.

fanquake commented on doc/build-netbsd.md:
```diff
@@ -1,92 +1,144 @@
-    BOOST_CPPFLAGS="-I/usr/pkg/include" \
-    BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx-4.8" \
-    BDB_CFLAGS="-I${BDB_PREFIX}/include" \
+    --with-boost-libdir=/usr/pkg/lib \
```
Is using BOOST_CPPFLAGS not sufficient? We no-longer link against any Boost libraries.

jarolrod replied:
actually i confirmed the only thing we now need is CPPFLAGS

fanquake commented on doc/build-netbsd.md:
```diff
@@ -1,92 +1,144 @@
+
+### 2. Clone Bitcoin Repo
+
+Now that `git` and the required dependencies are installed, let's clone the Bitcoin Core repository to a directory. All build scripts and commands will run from this directory.
```
```suggestion
Clone the Bitcoin Core repository to a directory. All build scripts and commands will run from this directory.
```

fanquake reviewed (commented):
concept ACK

jarolrod commented:
updated from d43fc74 to d3e9a1c ([pr25549.01](https://github.com/jarolrod/bitcoin/tree/pr25549.01) -> [pr25549.02](https://github.com/jarolrod/bitcoin/tree/pr25549.02), [diff](https://github.com/jarolrod/bitcoin/compare/pr25549.01..pr25549.02))

Addressed @fanquake comments: thanks for the review, I feel it is now more concise and to the point

jarolrod commented on doc/build-netbsd.md:
```diff
@@ -1,92 +1,144 @@
 
+##### Note on GCC
+
+We currently require GCC 8.1 or greater to compile; NetBSD currently ships with an older version. You should upgrade your GCC and then pass this new version to the configure script.
```
moved to `1. Install required dependencies`, also removed reference to any exact version of gcc we require. It's only needed to state that netbsd currently ships with a version which is less than is needed.

aureleoules reviewed (commented):
ACK d3e9a1c71bef1d730b5820f85e9758af54267ac3.
I was able to compile Bitcoin Core (with GUI) on a NetBSD 9.2 virtual machine using jarolrod's guide and followed the build guide without any issue.

fanquake reviewed (approved):
ACK d3e9a1c71bef1d730b5820f85e9758af54267ac3
